homeassistant/components/group/sensor.py

```diff
@@ -0,0 +1,405 @@
+                if state.last_updated > last_updated:
+                    self.last = float(state.state)
+                    self.last_entity_id = entity_id
+
+        # Set group as unavailable if all members are unavailable or missing
+        self._attr_available = any(state != STATE_UNAVAILABLE for state in states)
+
+        valid_state = self.mode(
+            state not in (STATE_UNKNOWN, STATE_UNAVAILABLE) for state in states
+        )
+        valid_state_numeric = self.mode(numeric_state for numeric_state in test_states)
+
+        if not valid_state or not valid_state_numeric:
+            # Set as unknown if any / all member is unknown or unavailable
+            self.min_entity_id = None
+            self.max_entity_id = None
+            self.last_entity_id = None
+            self._attr_native_value = None
+            return
+
+        # Calculate values
+        self._calculate_entity_properties()
+        self._calc_values(sensor_values)
+        self._attr_native_value = getattr(self, self._sensor_attr)
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the state attributes of the sensor."""
+        attributes: dict[str, Any] = {ATTR_ENTITY_ID: self._entity_ids}
+        if self._sensor_type == "min":
+            attributes[ATTR_MIN_ENTITY_ID] = self.min_entity_id
+        if self._sensor_type == "max":
+            attributes[ATTR_MAX_ENTITY_ID] = self.max_entity_id
+        if self._sensor_type == "last":
+            attributes[ATTR_LAST_ENTITY_ID] = self.last_entity_id
+        return attributes
+
+    @callback
+    def _calc_values(self, sensor_values: list[tuple[str, Any]]) -> None:
+        """Calculate the values."""
+
+        self.min_entity_id, self.min_value = calc_min(sensor_values)
+        self.max_entity_id, self.max_value = calc_max(sensor_values)
+        self.mean = calc_mean(sensor_values, self._round_digits)
+        self.median = calc_median(sensor_values, self._round_digits)
+        self.range = calc_range(sensor_values, self._round_digits)
+        self.sum = calc_sum(sensor_values, self._round_digits)
+
+    @property
+    def device_class(self) -> SensorDeviceClass | None:
+        """Return device class."""
+        if self._attr_device_class is not None:
+            return self._attr_device_class
+        return self.calc_device_class
+
+    @property
+    def state_class(self) -> SensorStateClass | str | None:
+        """Return state class."""
+        if self._attr_state_class is not None:
+            return self._attr_state_class
+        return self.calc_state_class
+
+    @property
+    def native_unit_of_measurement(self) -> str | None:
+        """Return native unit of measurement."""
+        if self._attr_native_unit_of_measurement is not None:
+            return self._attr_native_unit_of_measurement
+        return self.calc_unit_of_measurement
+
+    def _calculate_entity_properties(self) -> None:
```

Comment:
We should be lazy here and not calculate things which will never be used.
For example, if user has specified a device class, we should not calculate it.

Reply:
See if my changes is what you would expect

Reply:
Looks good 👍